### ISSUE TITLE
reprogramming atu-100

### ISSUE THREAD

**Issue by olson339** [Dfinitski/N7DDC-ATU-100-mini-and-extended-boards#22]
Need some help with the i2c reprogramming. I have a 1602 LCD with a i2c backpack. I used i2c scanner to find out the address, which is 3F. I changed the address using a PIC 3 and wrote the change to my ATU-100. Connected the LCD, background light comes on along with 16 square blocks on second row. I readjust the brightness but no readable display. I am using ver 3.0 of your frimware

**Comment by Dfinitski** [comment]
Hi.
Your I2C address scanner gives you 7 bit address, but the tuner board uses 8 bit address. 
As you can reed in the manual,
"**The port extension board for display 1602 made on the PCF8574AT chip has the address 7E, on the chip with the “T” index - 4E.** "
So no any ambivalent understanding which address you have to use.

**Comment by olson339** [comment]
so the scanner shows 0x3F. I enter 3F in cell 00 of the EEPROM. Should the entry be different

**Comment by Dfinitski** [comment]
Did you read my previous message ?
What kind of chip is used in the "backpack" of LCD ?

**Comment by olson339** [comment]
yes I read your previous message - the chip on the backpack is PCF8574AT

**Comment by Dfinitski** [comment]
Excellent, your I2C address is 7E.

**Comment by olson339** [comment]
ok thanks

**Comment by olson339** [comment]
The tuner is working but I haven't added any buttons yet for auto/bypass

**Comment by Dfinitski** [comment]
Can you confirm this behavior ?
https://www.youtube.com/watch?v=mUfD4gnzswc&t=1s

**Comment by olson339** [comment]
I ran the tuner using my 7300. Display worked correctly from star, tune, operate

**Comment by Dfinitski** [comment]
Thank you for the help.
you can test the newest firmware https://drive.google.com/file/d/1sCYbo3DPGjM-r94aqZJpuZqGIvasgv6K/view
if you want.

**Comment by valantis-chios** [comment]
Hi, I would like to inform you that the DIY kit that I bought has the same problem as described in the video. Firmware is 3.0. what do you propose to do? I am using lcd 16x2. I am waiting to receive pic programmer to check the hex and eeprom.
Thanks in advance

**Comment by Dfinitski** [comment]
Hi, 
the 3.1 firmware version should work fine. Use a Pickit 3 to change the old firmware.
If you will have some problems with 1602 LCD please tell me know ASAP.

**Comment by valantis-chios** [comment]
20201113_161618.mp4
<https://drive.google.com/file/d/1aXIEYz6RTYQtXtyHNyLcT5btIjO8TpyP/view?usp=drivesdk>

  20201113_161849.mp4
<https://drive.google.com/file/d/1elP15rnoYLPoujuj1u5GEO6MsJs92JkB/view?usp=drivesdk>



Στις Τετ, 4 Νοε 2020 - 18:33 ο χρήστης David Fainitski <
notifications@github.com> έγραψε:

**Comment by Dfinitski** [comment]
Hi,
What kind of chip is used in the I2C extender of your LCD display ?

**Comment by valantis-chios** [comment]
Hi, the chip is PCF8574T

**Comment by Dfinitski** [comment]
Thanks.
I will order LCD with this chip and check this by myself.
I can not find the problem in the code also other users report about good
work with LCD.

**Comment by valantis-chios** [comment]
One more thing,  how does the type of display affect the tuning process?

**Comment by Dfinitski** [comment]
I is not matter, you can use the tuner even with no any displays, it is
only an option.

**Comment by valantis-chios** [comment]
20201113_235455.mp4
<https://drive.google.com/file/d/1IMidnI6zYtDpbtd9VnnCxdzalz9Px1o9/view?usp=drivesdk>


Στις Παρ, 13 Νοε 2020 - 21:04 ο χρήστης David Fainitski <
notifications@github.com> έγραψε:

**Comment by valantis-chios** [comment]
Hi, I rewind tandem match and the problem with tuning seems to be solved ( maybe there was a short between filament and ferrite). But the problem with no swr indication  still exist.

**Comment by Dfinitski** [comment]
The LCD will come to me next Thursday. I will see.

пт, 13 нояб. 2020 г. в 14:44, valantis-chios <notifications@github.com>:

**Comment by Dfinitski** [comment]
Hi.
I have bouth randomly an LCD with PCB8574T expansion board on Ebay, I2C address is 4E.
Everithing works good. 
I tried to find mistakes in the code also, many times, but I don't see anything wrong.

**Comment by Dfinitski** [comment]
People, watch this video https://www.youtube.com/watch?v=vWtbs4n4ewo&t=18s
If somebody knows what is going on - tell me know, I will fix it.

**Comment by valantis-chios** [comment]
Hi, I have the same problem like no 2 and no 3 lcd( no swr indication). Tuning possess is ok.

**Comment by Dfinitski** [comment]
Hi. Order the chiepest LCD on Ebay or Aliexpress, should to work. I don;t know what is the reason, so I can't resolve this problem. 
I don't have a strong expirience with this kind LCD, i have no idea what is going on. But definatelly those LCD have some differences.

**Comment by owenaggi** [comment]
help i dont understand what i have done i have the picket 3 and ext board i wanted to use it on low power when trying to reprogram i have some how deletet the pic and now when i try to program the pic picket 3 tells me that there is insuffichet memory please tell me a what i did wrong and if possable how to fix it

**Comment by TA2KW** [comment]
hello, I made a connection with 1602 LCD, I2C PCF8574T converter, but the 7805 transistor in the circuit supply is overheating, I could not solve this problem, why does the 7805 get hot? :(

**Comment by Dfinitski** [comment]
Hi.
Everything works good but 7805 is hot ?
What is the voltage rate of the relays ?

**Comment by TA2KW** [comment]
1602 LCD screen gets hot when I plug it in,
No problem with oled display.

**Comment by Dfinitski** [comment]
So check the connection, power voltage polarity.
